python-threatexchange/threatexchange/exchanges/impl/fb_threatexchange_api.py

```diff
@@ -321,19 +337,117 @@ def report_false_positive(
             ),
         )
 
+    @classmethod
+    def naive_convert_to_signal_type(
+        cls,
+        signal_types: t.Sequence[t.Type[SignalType]],
+        fetched: t.Dict[t.Tuple[str, str], t.Optional[FBThreatExchangeIndicatorRecord]],
+    ) -> t.Dict[t.Type[SignalType], t.Dict[str, FBThreatExchangeIndicatorRecord]]:
+        ret: t.Dict[
+            t.Type[SignalType], t.Dict[str, FBThreatExchangeIndicatorRecord]
+        ] = {}
+        mapping = _make_indicator_type_mapping(signal_types)
+
+        for (type_str, signal_str), metadata in fetched.items():
+            potential_types = mapping.get(type_str)
+            if potential_types is None or metadata is None:
+                continue
+            indicator_tags = {t for opinion in metadata.opinions for t in opinion.tags}
+            for tag, s_types in potential_types.items():
+                if tag is not None and tag not in indicator_tags:
+                    continue
+                for tx_s_type in s_types:
+                    s_type_specific_signal_str = (
+                        tx_s_type.normalize_fb_threatexchange_indicator(
+                            type_str, signal_str, tag
+                        )
+                    )
+                    s_type = t.cast(t.Type[SignalType], tx_s_type)
+                    inner = ret.get(s_type)
+                    if inner is None:
+                        inner = {}
+                        ret[s_type] = inner
+                    to_insert = _merge_record_for_signal_type(
+                        metadata, tag, inner.get(s_type_specific_signal_str)
+                    )
+                    if to_insert is not None:
+                        inner[s_type_specific_signal_str] = to_insert
+
+        return ret
+
+
+def _merge_record_for_signal_type(
+    tx_record: FBThreatExchangeIndicatorRecord,
+    tag: t.Optional[str],
+    existing: t.Optional[FBThreatExchangeIndicatorRecord],
+) -> t.Optional[FBThreatExchangeIndicatorRecord]:
+    if tag is not None:
+        applicable_opinions = [
+            o for o in tx_record.opinions if any(t in tag for t in o.tags)
+        ]
+        if not applicable_opinions:
+            return None
+        if len(applicable_opinions) != len(tx_record.opinions):
+            tx_record = FBThreatExchangeIndicatorRecord(applicable_opinions)
+    if existing is not None:
+        existing.merge(tx_record)
+        return None
+    return tx_record
+
+
+def _indicator_applies(
+    u: ThreatUpdateJSON,
+    type_mapping: t.Mapping[
+        str,
+        t.Mapping[
+            t.Optional[str], t.Sequence[t.Type[HasFbThreatExchangeIndicatorType]]
+        ],
+    ],
+) -> t.Optional[FBThreatExchangeIndicatorRecord]:
+    """Based on the available signal types, return a record"""
+    potential_signal_type = type_mapping.get(u.threat_type)
+    if potential_signal_type is None:
+        return None
```

Comment:
I'd guess no, because it's a normal part of operation when you've selected that you only want certain signal types. I've so far avoided logging on individual calls to ThreatExchange because the sheer volume of records (50k + warns for a full fetch just from TMK).

I could see a world in which we used logging filters to allow you to selectively enable certain file logging, and have them off by default, but have been avoiding nice-to-haves for need-to-haves